src/bosh.js

```diff
@@ -391,6 +391,11 @@ Strophe.Bosh.prototype = {
                      ", number of errors: " + this.errors);
         if (this.errors > 4) {
             this._conn._onDisconnectTimeout();
+        } else {
+          var err_callback = this._conn.httpErrorHandler[reqStatus];
+          if(err_callback){
+            err_callback.apply(this, [reqStatus,]);
+          }
         }
```

Comment:
Please stick to the style used in Strophe. We indent 4 spaces, not 2 and there should be a space between `if` and the brackets `(err_callback)`.